core/src/main/java/io/ably/lib/realtime/Presence.java

```diff
@@ -1031,13 +1038,47 @@ synchronized void clear() {
                 residualMembers.clear();
         }
 
+        /**
+         * Spec: RTP17g
+         */
+        synchronized void reEnter() {
+            for (Map.Entry<String, PresenceMessage> entry: members.entrySet()) {
+                PresenceMessage member = entry.getValue();
+                member.action = PresenceMessage.Action.enter;
+                try {
+                    updatePresence(member, null);
+                } catch (AblyException e) {
+                    String errorString = String.format(Locale.ROOT, "Cannot automatically re-enter %s on channel %s (%s)",
+                        member.clientId, channel.name, e.errorInfo.message);
+                    Log.e(TAG, errorString);
```

Comment:
As per example provided here ->
 https://github.com/ably/ably-java/pull/842#discussion_r1014099918
We might wanna emit error as a update
`channel.emitUpdate(new ErrorInfo(errorString, 91004), true);`
Just check if right errorCode used to emit the error.
I think all error codes are mentioned here ->
https://github.com/ably/ably-common/blob/main/protocol/errors.json
if error is not channel specific, rather connection specific, you can find there must be a way to emit error on connection
`connection.emitUpdate(new ErrorInfo(errorString, 91004), true);`

Reply:
@sacOO7 I think this is it https://github.com/ably/ably-java/commit/d022c3d5013d993f5c0bd49a0ec7a2c53e38fc46

Reply:
Looks good